libs/sdk-core/src/breez_services.rs

```diff
@@ -1896,26 +1896,49 @@ impl Receiver for PaymentReceiver {
     }
 }
 
-/// Convenience method to look up LSP info based on current LSP ID
-async fn get_lsp(persister: Arc<SqliteStorage>, lsp: Arc<dyn LspAPI>) -> Result<LspInformation> {
+/// Convenience method to look up LSP info based on current LSP ID and
+/// fallback to the first LSP in the listed LSPs if the current LSP is not available.
+async fn get_lsp(
+    persister: Arc<SqliteStorage>,
+    lsp_api: Arc<dyn LspAPI>,
+) -> Result<LspInformation> {
     let lsp_id = persister.get_lsp_id()?.ok_or(anyhow!("No LSP ID found"))?;
```

Comment:
Shouldn't we continue here and try to connect to the first lsp from the list?

Reply:
Wouldn't it always be set from the config default_lsp_id?... Ah, it could be that it's also set to None right